integrations/opsgenie/main.py

```diff
@@ -16,48 +19,61 @@ def init_client() -> OpsGenieClient:
     )
 
 
-async def enrich_services_with_team_data(
+async def enrich_services_with_oncall_data(
     opsgenie_client: OpsGenieClient,
     semaphore: asyncio.Semaphore,
-    service: dict[str, Any],
-) -> dict[str, Any]:
-    async with semaphore:
-        team_data, schedule = await asyncio.gather(
-            opsgenie_client.get_oncall_team(service["teamId"]),
-            opsgenie_client.get_schedule_by_team(service["teamId"]),
-        )
+    service_batch: list[dict[str, Any]],
+) -> list[dict[str, Any]]:
 
-        service["__team"] = team_data
-        if schedule:
-            service["__oncalls"] = await opsgenie_client.get_oncall_user(schedule["id"])
-        return service
+    team_ids = {service["teamId"] for service in service_batch if service.get("teamId")}
 
+    schedules = await opsgenie_client.get_schedules_by_teams(list(team_ids))
 
-async def enrich_incident_with_alert_data(
-    opsgenie_client: OpsGenieClient,
-    semaphore: asyncio.Semaphore,
-    incident: dict[str, Any],
-) -> dict[str, Any]:
-    async with semaphore:
-        if not incident["impactedServices"]:
-            return incident
-        impacted_services = await opsgenie_client.get_impacted_services(
-            incident["impactedServices"]
-        )
-        incident["__impactedServices"] = impacted_services
-        return incident
+    schedule_ids = [schedule["id"] for schedule in schedules.values()]
 
+    async def fetch_oncall(schedule_id: str) -> tuple[str, dict[str, Any]]:
+        async with semaphore:
+            return schedule_id, await opsgenie_client.get_oncall_user(schedule_id)
```

Comment:
can't we solve this by some solution on port side, like calculation property or something? I would like to try to reduce this kind of extra requests

Reply:
I wonder if this can be a separate kind, something like service-on-call or something like that